### PR TITLE
Empty strings as ANY_VALUE + Handle dynamic blocks

### PR DESCRIPTION
**Pull request by rotemavni**
This PR has two improvements:

**Stop identifying empty strings as ANY_VALUE**
Until now, if a check had `ANY_VALUE` as the expected value, if the value is `""` the check passed, while it should have failed.
I added a validation that if the value is a string and it is empty, fail the check.
Added tests for it, under `test_SQSQueueEncryption`.

**Add terraform dynamic blocks to the graph vertices**
I extracted `handle_dynamic_values` from `base_resource_check` to `parser_functions` and use it when creating the resources graph vertices.
Added a test for it in `test_graph_builder`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/cloudformation/checks/resource/aws/example_SQSQueueEncryption/test_SQSQueueEncryption-FAILED.yml [hunk 1]
```diff
@@ -5,4 +5,5 @@ Resources:
     Properties:
       RedrivePolicy:
         deadLetterTargetArn: "example_arn"
-        maxReceiveCount: 5
\ No newline at end of file
+        maxReceiveCount: 5
+      KmsMasterKeyId: ""
```
Comment (gruebel) [inline tests/cloudformation/checks/resource/aws/example_SQSQueueEncryption/test_SQSQueueEncryption-FAILED.yml:9]: didn't know that this actually a valid configuration 😱

File: checkov/terraform/parser_functions.py [hunk 2]
```diff
@@ -167,3 +168,28 @@ def _check_map_type_consistency(value: Dict) -> Dict:
...
+
+
+def handle_dynamic_values(conf: Dict[str, List[Any]]) -> None:
+    # recursively search for blocks that are dynamic
+    for block_name in conf.keys():
+        if isinstance(conf[block_name], dict):
+            handle_dynamic_values(conf[block_name])
+
+        # if the configuration is a block element, search down again.
+        if isinstance(conf[block_name], list) and len(conf[block_name]) > 0 and isinstance(conf[block_name][0], dict):
```
Comment (gruebel) [inline checkov/terraform/parser_functions.py:180]:
```suggestion
        if isinstance(conf[block_name], list) and conf[block_name] and isinstance(conf[block_name][0], dict):
```
no need to calculate the length, if you just want to know, it is not empty. this way is also faster 🚀

File: checkov/terraform/checks/resource/base_resource_value_check.py [hunk 3]
```diff
@@ -55,15 +56,15 @@ def _is_nesting_key(inspected_attributes: List[str], key: List[str]) -> bool:
...
+        handle_dynamic_values(conf)
         inspected_key = self.get_inspected_key()
         expected_values = self.get_expected_values()
         if dpath.search(conf, inspected_key) != {}:
             # Inspected key exists
             value = dpath.get(conf, inspected_key)
             if isinstance(value, list) and len(value) == 1:
                 value = value[0]
-            if ANY_VALUE in expected_values and value is not None:
+            if ANY_VALUE in expected_values and value is not None and (not isinstance(value, str) or len(value) > 0):
```
Comment (gruebel) [inline checkov/terraform/checks/resource/base_resource_value_check.py:67]:
```suggestion
            if ANY_VALUE in expected_values and value is not None and (not isinstance(value, str) or value):
```
same reason

File: checkov/cloudformation/checks/resource/base_resource_value_check.py [hunk 4]
```diff
@@ -66,10 +66,10 @@ def scan_resource_conf(self, conf: Dict[StrNode, DictNode]) -> CheckResult:
 
                 if match[:-1] == path_elements:
                     # Inspected key exists
-                    if ANY_VALUE in expected_values:
+                    value = match[-1]
+                    if ANY_VALUE in expected_values and value is not None and (not isinstance(value, str) or len(value) > 0):
```
Comment (gruebel) [inline checkov/cloudformation/checks/resource/base_resource_value_check.py:70]:
```suggestion
                    if ANY_VALUE in expected_values and value is not None and (not isinstance(value, str) or value):
```
same reason

**Review by gruebel — COMMENTED** [review]
👍

**Review by tronxd — APPROVED** [review]

File: tests/cloudformation/checks/resource/aws/example_SQSQueueEncryption/test_SQSQueueEncryption-FAILED.yml (hunk 1 above)
Comment (rotemavni) [reply tests/cloudformation/checks/resource/aws/example_SQSQueueEncryption/test_SQSQueueEncryption-FAILED.yml:9]: Perhaps it's not 🤷‍♀️  I just wanted to validate the string check works for cloudformation too

**Pull request merged**